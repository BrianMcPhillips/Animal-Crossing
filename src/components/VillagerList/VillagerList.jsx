import React from 'react';
import PropTypes from 'prop-types';
import Villager from './Villager/Villager';
import styles from './VillagerList.module.css';

const VillagerList = ({ villagers }) => {
  const villagerElements = villagers.map(villager => (
    <li key={villager._id}>
      <Villager {...villager} /> 
    </li>
  ));
  return (
    <ul data-testid="villagers" className={styles.VillagerList}>
      {villagerElements}
    </ul>
  );
};

VillagerList.propTypes = {
  villagers: PropTypes.arrayOf(PropTypes.shape({
    _id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired
  })).isRequired,
};

export default VillagerList;
