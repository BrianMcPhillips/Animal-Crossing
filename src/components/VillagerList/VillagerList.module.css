.VillagerList {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  align-items: end;
  list-style-type: none;
}
