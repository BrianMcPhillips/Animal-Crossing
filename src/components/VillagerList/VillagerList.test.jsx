import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VillagerList from './VillagerList';

describe('VillagerList component', () => {
  afterEach(() => cleanup());
  it('renders a Villager List', () => {
    const { asFragment } = render(<VillagerList 
      villagers={[
        { 
          _id: '1234', 
          name: 'MyVillager', 
          imageUrl: 'http://placeholder.com/villager' 
        }
      ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
