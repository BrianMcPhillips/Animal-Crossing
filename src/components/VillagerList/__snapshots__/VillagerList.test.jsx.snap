// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VillagerList component renders a Villager List 1`] = `
<DocumentFragment>
  <ul
    class="VillagerList"
    data-testid="villagers"
  >
    <li>
      <figure
        class="Villager"
      >
        <img
          alt="MyVillager"
        />
        <figcaption>
          MyVillager
        </figcaption>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;
